lib/origen/core_ext/range.rb

```diff
@@ -13,4 +13,8 @@ def to_a
       a
     end
   end
+
+  def <=>(other)
+    [min, max] <=> [other.min, other.max]
+  end
```

Comment:
This seems a bit heavy handed, how do ranges normally compare to each other?

Reply:
@ginty maybe `(min..max).eql?(Range.new(other.min, other.max))` ? not sure if this has some ruby version dependencies

Reply:
Dunno, but this seems to dodgy to me, overriding a primary behavior of a core Ruby class just to implement an underscore escaper.
I think this needs to be removed and add a method somewhere to do what you need.

Reply:
> This seems a bit heavy handed, how do ranges normally compare to each other?

Ranges do not sort based on [this discussion](https://stackoverflow.com/questions/18498738/sort-an-array-of-ranges).